tools/omnissm/host/register/initialize.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2018 Capital One Services, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// SSMAgentRegistration Sourced from /var/lib/amazon/ssm/registration
+type SSMAgentRegistration struct {
+	ManagedInstanceID string
+	Region            string
+}
+
+// LinuxSSMRegistrationPath Linux Path to Agent Registration State
+const LinuxSSMRegistrationPath = "/var/lib/amazon/ssm/registration"
+
+// IdentityURL ec2 metadata server instance identity document
+const IdentityURL = "http://169.254.169.254/latest/dynamic/instance-identity/document"
+
+// SignatureURL RSA SHA256 Signature of identity document
+const SignatureURL = "http://169.254.169.254/latest/dynamic/instance-identity/signature"
+
+// FetchContents Retrieve contents of URL
+func FetchContents(uri string) ([]byte, error) {
+	response, err := http.Get(uri)
```

Comment:
Later a custom `http.Client` is used, I would consider embedding that into a struct and adding this as a method so the same `http.Client` can easily be used for all http communications.

Reply:
yeah the custom http client was for off box communications timeout enforcement, the local metadata server endpoint  is low latency (although low concurrency).